library/common/extensions/key_value/platform/config.h

```diff
@@ -0,0 +1,62 @@
+#include "envoy/common/key_value_store.h"
+
+#include "source/common/common/key_value_store_base.h"
+
+#include "library/common/extensions/key_value/platform/platform.pb.h"
+#include "library/common/extensions/key_value/platform/platform.pb.validate.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace KeyValue {
+
+class PlatformInterface {
+public:
+  virtual ~PlatformInterface() {}
+  virtual void flush(const std::string& key, const std::string& contents) PURE;
```

Comment:
nit: I would suggest one of `store`, `save`, `write` or `put` as opposed to `flush` probably.